### DISCUSSION TITLE
How to `provider-cloudflare`?

### DISCUSSION THREAD

**Discussion by milkpirate** [crossplane/upjet#443] [Q&A]
### What problem are you facing?
I would like to generate a crossplane provider from [terraform-provider-cloudflare](https://github.com/cloudflare/terraform-provider-cloudflare). There already exists [one](https://github.com/cdloh/provider-cloudflare) which was generated from v3.31.0 of the TF provider, but since I need some features from >v4.0.0 and @cdloh does seem to have the capacity for an update I gave it a shot myself and followed [generating a Crossplane provider](https://github.com/crossplane/upjet/blob/main/docs/generating-a-provider.md).

But then @cdloh told me, its not as easy as this, since the TF provider in v4 is using a different (TF SDKv2 I guess) API version (see https://github.com/cdloh/provider-cloudflare/issues/54). He also said I should have a look at https://github.com/crossplane-contrib/provider-upjet-aws/blob/main/config/externalname.go, but its absolutely not clear to me whats important and what not.

@cdloh also said there is not much documentation on that and that is why I am here.

### How could Upjet help solve your problem?
Please extend the [generating a Crossplane provider](https://github.com/crossplane/upjet/blob/main/docs/generating-a-provider.md) manual to accommodate the what the general steps are to get a provider such as the TF cloudflare one working as a source for upjet.

**Comment by NerdJeremia** [comment]
Hi @milkpirate,
we used upjet on a terraform provider that was built with the latest [terraform plugin framework](https://developer.hashicorp.com/terraform/plugin/framework) version.
We did not encounter any problems with that so you should be good just following the upjet docs.
